src/commands/command-executor.js

```diff
@@ -449,6 +472,12 @@ class CommandExecutor {
 
         const commands = [];
         for (const command of pendingCommands) {
+            if (command?.isBlocking === true && command?.parentId === null) {
```

Comment:
You don't need to check if boolean is equal to true, you can just use isBlocking && ...

Reply:
Imo we should have stricter conditions, truthy conditions may cause issues that are hard to debug